tests/ui/confuse-field-and-method/issue-2392.rs

```diff
@@ -1,3 +1,7 @@
+// FIXME(estebank): diagnostics with long type paths that don't print out the full path anywhere
+// still prints the note explaining where the type was written to.
+//@ compile-flags: -Zwrite-long-types-to-disk=yes
```

Comment:
I found these by changing the default `-Zwrite-long-types-to-disk=no` in `compiletest::runtest::TestCx::make_compile_args` to `yes`, which highlighted some preexisting cases that I'll investigate.

Reply:
One left: `tests/ui/methods/filter-relevant-fn-bounds.rs`. Also pre-existing.